source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -780,10 +799,21 @@ void InjaTransformer::transform(Http::RequestOrResponseHeaderMap &header_map,
   for (const auto &templated_dynamic_metadata : dynamic_metadata_) {
     std::string output = instance_->render(templated_dynamic_metadata.template_);
     if (!output.empty()) {
-      ProtobufWkt::Struct strct(
-          MessageUtil::keyValueStruct(templated_dynamic_metadata.key_, output));
-      callbacks.streamInfo().setDynamicMetadata(
-          templated_dynamic_metadata.namespace_, strct);
+      // Need to check if number
```

Comment:
this breaks backwards compatbility; if customers have number like strings today, we will now change them to a number implicity. is there a way to explicitly ask for a number?